### PR TITLE
Content portal tabs/fullcheck

### PR DESCRIPTION
**Pull request by cnevinc**
ready for review

### CODE REVIEW

File: app/src/main/java/org/mozilla/rocket/content/ContentPortalView.kt [hunk 1]
```diff
@@ -150,6 +178,19 @@ class ContentPortalView : CoordinatorLayout, ContentPortalListener {
         visibility = VISIBLE
         // TODO: if previously is collapsed, collapse here.
         bottomSheetBehavior?.state = BottomSheetBehavior.STATE_EXPANDED
+
+        // content fragment will be added every time when it's displayed. Since back press
+        // will always remove it from fragment manager
+        if (features().size != 1 && !AppConstants.isProductionBuild()) {
+            initContentFragment()
```
Comment (mTwTm) [inline app/src/main/java/org/mozilla/rocket/content/ContentPortalView.kt]: Does it make sense that initContentFragment() is delayed but initLegacyContentPortal() is not?

File: app/src/main/java/org/mozilla/rocket/content/ContentPortalView.kt [hunk 2]
```diff
@@ -308,5 +338,11 @@ class ContentPortalView : CoordinatorLayout, ContentPortalListener {
...
         private const val NEWS_THRESHOLD = 10
         private const val SHOPPINGLINK_GRID_SPAN = 2
+        private const val TAG_CONTENT_FRAGMENT = "TAG_CONTENT_FRAGMENT"
     }
-}
\ No newline at end of file
+}
+
+// helper method to work with FragmentManager
+inline fun Context.inTransaction(func: FragmentTransaction.() -> FragmentTransaction) {
```
Comment (mTwTm) [inline app/src/main/java/org/mozilla/rocket/content/ContentPortalView.kt]: Consider place this somewhere else.

File: app/src/main/java/org/mozilla/focus/home/HomeFragment.java [hunk 3]
```diff
@@ -330,7 +328,7 @@ private void initHomeBanner(Context context) {
     }
 
     private void initCouponBanner(Context context) {
-        initBanner(context, AppConfigWrapper.getCouponBannerRootConfig(), CURRENT_COUPON_BANNER_CONFIG, bannerConfigViewModel.getCouponConfig(), contentPanel::hideCouponBanner);
+        initBanner(context, AppConfigWrapper.getCouponBannerRootConfig(), CURRENT_COUPON_BANNER_CONFIG, bannerConfigViewModel.getCouponConfig(), null);
```
Comment (mTwTm) [inline app/src/main/java/org/mozilla/focus/home/HomeFragment.java]: consider removing the least parameter, the purpose of it was solely to delegate callback

File: app/build.gradle [hunk 4]
```diff
@@ -52,6 +52,10 @@ android {
             sourceCompatibility JavaVersion.VERSION_1_8
             targetCompatibility JavaVersion.VERSION_1_8
         }
+
+        kotlinOptions {
+            jvmTarget = JavaVersion.VERSION_1_8
```
[the remark this answers is not in the document]
Comment (cnevinc) [reply app/build.gradle]: Not every module has Kotlin. So it'll be batter let the library module define it.

File: app/src/main/java/org/mozilla/focus/home/HomeFragment.java (hunk 3 above)
Comment (cnevinc) [reply app/src/main/java/org/mozilla/focus/home/HomeFragment.java]: HomeBanner do need `hideHomeBannerProcedure`, so it's better to keep it.

File: app/src/main/java/org/mozilla/rocket/content/ContentPortalView.kt [hunk 5]
```diff
@@ -63,40 +66,65 @@ class ContentPortalView : CoordinatorLayout, ContentPortalListener {
...
-            setupViewNews()
+            features.add(TYPE_COUPON)
+        }
+
+        if (AppConfigWrapper.hasEcommerceShoppingLink()) {
+            features.add(TYPE_TICKET)
+        }
+
+        if (AppConfigWrapper.hasNewsPortal()) {
+            // features.add(TYPE_NEWS) // we don't support News in Tab now.
```
[the remark this answers is not in the document]
Comment (cnevinc) [reply app/src/main/java/org/mozilla/rocket/content/ContentPortalView.kt]: oh... I think I've made a mistake. I should still allow TYPE_NEWS here. Cause I'll remove it in the `adjust()` method

File: app/src/main/java/org/mozilla/rocket/content/ContentPortalView.kt (hunk 1 above)
Comment (cnevinc) [reply app/src/main/java/org/mozilla/rocket/content/ContentPortalView.kt]: yes. `initContentFragment ()` needs to be called **every time** when we display the ContentPortalView.
While  "initLegacyContentPortal" will inflate the ContentPortalView and its content with HomeFragment. So we don't need to add it again.

Comment (cnevinc) [reply app/src/main/java/org/mozilla/rocket/content/ContentPortalView.kt]: fixed

File: app/src/main/java/org/mozilla/rocket/content/ContentPortalView.kt (hunk 2 above)
Comment (cnevinc) [reply app/src/main/java/org/mozilla/rocket/content/ContentPortalView.kt]: fixed

**Comment by cnevinc** [comment]
> ing commit to turn on this feature in production. Before that, we should not merge this PR into master

I'd like to land this and only enable it in Nightly.
While QA verifying in Nightly, I'll add some tests before it goes live.

File: app/src/main/res/layout/content_tab_coupon.xml [hunk 6]
```diff
@@ -31,7 +31,7 @@
             app:layout_constraintBottom_toBottomOf="parent"
             app:layout_constraintEnd_toEndOf="parent"
             app:layout_constraintStart_toStartOf="parent"
-            app:layout_constraintTop_toBottomOf="@id/content_banners"/>
+            app:layout_constraintTop_toBottomOf="@id/content_banners "/>
```
Comment (benjamin-cheng) [inline app/src/main/res/layout/content_tab_coupon.xml]: Space was inserted in the wrong location. wonder why it can be compiled? However, lint catches it.

**Pull request merged**